cloudify_cli/replace_certificates_config.py

```diff
@@ -0,0 +1,377 @@
+                                     new_cert_path, new_key_path,
+                                     new_ca_cert_path, logger)
+
+    def _create_paths_mapping(self):
+        return {
+            'new_postgresql_cert_path': self.new_cert_path,
+            'new_postgresql_key_path': self.new_key_path,
+            'new_postgresql_ca_cert_path': self.new_ca_cert_path
+        }
+
+
+class ManagerNode(Node):
+    def __init__(self,
+                 host_ip,
+                 username,
+                 key_file_path,
+                 new_internal_cert_path,
+                 new_internal_key_path,
+                 new_external_cert_path,
+                 new_external_key_path,
+                 new_postgresql_client_cert_path,
+                 new_postgresql_client_key_path,
+                 new_ca_cert_path,
+                 new_external_ca_cert_path,
+                 new_postgresql_ca_cert_path,
+                 new_ldap_ca_cert_path,
+                 logger):
+        self.new_external_cert_path = new_external_cert_path
+        self.new_external_key_path = new_external_key_path
+        self.new_external_ca_cert = new_external_ca_cert_path
+        self.new_postgresql_client_cert_path = new_postgresql_client_cert_path
+        self.new_postgresql_client_key_path = new_postgresql_client_key_path
+        self.new_postgresql_ca_cert_path = new_postgresql_ca_cert_path
+        self.new_ldap_ca_cert_path = new_ldap_ca_cert_path
+
+        super(ManagerNode, self).__init__(host_ip, username, key_file_path,
+                                          new_internal_cert_path,
+                                          new_internal_key_path,
+                                          new_ca_cert_path, logger)
+
+    def _create_paths_mapping(self):
+        return {
+            'new_internal_cert_path': self.new_cert_path,
+            'new_internal_key_path': self.new_key_path,
+            'new_ca_cert_path': self.new_ca_cert_path,
+            'new_external_cert_path': self.new_external_cert_path,
+            'new_external_key_path': self.new_external_key_path,
+            'new_external_ca_cert_path': self.new_external_ca_cert,
+            'new_postgresql_client_cert_path':
+                self.new_postgresql_client_cert_path,
+            'new_postgresql_client_key_path':
+                self.new_postgresql_client_key_path,
+            'new_postgresql_ca_cert_path': self.new_postgresql_ca_cert_path,
+            'new_ldap_ca_cert_path': self.new_ldap_ca_cert_path
+        }
+
+
+class ReplaceCertificatesConfig(object):
+    def __init__(self, config_dict, is_all_in_one, logger):
+        if SSHClient is None:
+            raise CloudifyCliError("SSH not available - fabric not installed")
+
+        self.logger = logger
+        self.is_all_in_one = is_all_in_one
+        self.config_dict = config_dict
+        self.username = env.profile.ssh_user
+        self.key_file_path = env.profile.ssh_key
```

Comment:
those can be hopefully made to use the cluster profile (in case of different names/keys per node)

Reply:
Correct :)